test/Spec/MockContract.hs

```diff
@@ -209,9 +210,19 @@ mockCallCommand ShellArgs {cmdName, cmdArgs, cmdOutParser} = do
   case (cmdName, cmdArgs) of
     ("cardano-cli", "query" : "utxo" : "--address" : addr : _) ->
       cmdOutParser <$> mockQueryUtxo addr
-    ("cardano-cli", "transaction" : "build" : _) ->
+    ("cardano-cli", "transaction" : "build" : args) -> do
+      case drop 1 $ dropWhile (/= "--out-file") args of
+        filepath : _ ->
+          modify (\st -> st {files = Map.insert (Text.unpack filepath) (OtherFile "TxBody") st.files})
```

Comment:
Looks like this could benefit from lenses and `modifying`, and just in general, if you're using a state monad, lenses are lovely.
Though, Im unsure if you get this luxury when using Freer...

Reply:
Addressed this, but I did not change to lenses for the getters, I thought the record dot syntax does the job well on that fron. What are your opinions about this?